### PR TITLE
Adding my DZ60, Satan, XD75 kemaps and updates to userspace

### PR DESCRIPTION
**Pull request by JarredSteenvoorden**
Adding keymaps for DZ60, Satan and XD75 (mine and a friends).

Also small update to userspace.

Cheers!

### CODE REVIEW

File: keyboards/dz60/keymaps/jarred/config.h [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#ifndef CONFIG_USER_H
```
Comment (drashna) [inline keyboards/dz60/keymaps/jarred/config.h]: If you could replace the include guard (the ifndef, define, and endif at the end) with just `#pragma once`? 

It's simpler, and less prone to user error, so if you could make this switch?

File: keyboards/dz60/keymaps/jarred/config.h [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+#define CONFIG_USER_H
+ 
+#include "../../config.h"
```
Comment (drashna) [inline keyboards/dz60/keymaps/jarred/config.h]: This line is not actually needed, and should be removed. 

Specifically, the build script automatically includes all of the config.h files, so this is not necessary.

File: keyboards/dz60/keymaps/jarred/config.h [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+ 
+#define PREVENT_STUCK_MODIFIERS
```
Comment (drashna) [inline keyboards/dz60/keymaps/jarred/config.h]: This is actually the new default now. This isn't needed.  But you may need to update your fork to get the changes.

File: keyboards/planck/keymaps/jarred/config.h [hunk 4]
```diff
@@ -19,6 +19,8 @@
 
 #include "config_common.h"
 
+<<<<<<< HEAD
```
Comment (drashna) [inline keyboards/planck/keymaps/jarred/config.h]: You have merge conflicts here, you need to fix this.

File: keyboards/satan/keymaps/jarred/config.h [hunk 5]
```diff
@@ -0,0 +1,8 @@
+#ifndef CONFIG_USER_H
```
Comment (drashna) [inline keyboards/satan/keymaps/jarred/config.h]: If you could replace the include guard (the ifndef, define, and endif at the end) with just `#pragma once`? 

It's simpler, and less prone to user error, so if you could make this switch?

File: keyboards/satan/keymaps/jarred/config.h [hunk 6]
```diff
@@ -0,0 +1,8 @@
...
+#define CONFIG_USER_H
+ 
+#include "../../config.h"
```
Comment (drashna) [inline keyboards/satan/keymaps/jarred/config.h]: This line is not actually needed, and should be removed. 

Specifically, the build script automatically includes all of the config.h files, so this is not necessary.

File: keyboards/satan/keymaps/jarred/config.h [hunk 7]
```diff
@@ -0,0 +1,8 @@
...
+ 
+#define PREVENT_STUCK_MODIFIERS
```
Comment (drashna) [inline keyboards/satan/keymaps/jarred/config.h]: This line is not actually needed, and should be removed.

Specifically, the build script automatically includes all of the config.h files, so this is not necessary.

File: keyboards/satan/keymaps/jarred/rules.mk [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+NKRO_ENABLE = yes           # Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = yes       # Enable keyboard backlight functionality
+MIDI_ENABLE = no            # MIDI controls
+AUDIO_ENABLE = no           # Audio output on port C6
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+RGBLIGHT_ENABLE = no        # Enable WS2812 RGB underlight.  Do not enable this with audio at the same time.
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+
+ifndef QUANTUM_DIR
```
Comment (drashna) [inline keyboards/satan/keymaps/jarred/rules.mk]: Could you remove this (ifndef/include/endif) block? 

It's not needed, and is a holdover from the old build system.

File: keyboards/xd75/keymaps/daniel/config.h [hunk 9]
```diff
@@ -0,0 +1,8 @@
+#ifndef CONFIG_USER_H
```
Comment (drashna) [inline keyboards/xd75/keymaps/daniel/config.h]: If you could replace the include guard (the ifndef, define, and endif at the end) with just `#pragma once`? 

It's simpler, and less prone to user error, so if you could make this switch?

File: keyboards/xd75/keymaps/daniel/config.h [hunk 10]
```diff
@@ -0,0 +1,8 @@
...
+#define CONFIG_USER_H
+
+#include "../../config.h"
```
Comment (drashna) [inline keyboards/xd75/keymaps/daniel/config.h]: This line is not actually needed, and should be removed. 

Specifically, the build script automatically includes all of the config.h files, so this is not necessary.

File: keyboards/xd75/keymaps/daniel/config.h [hunk 11]
```diff
@@ -0,0 +1,8 @@
...
+
+#define PREVENT_STUCK_MODIFIERS
```
Comment (drashna) [inline keyboards/xd75/keymaps/daniel/config.h]: This line is not actually needed, and should be removed.

Specifically, the build script automatically includes all of the config.h files, so this is not necessary.

File: keyboards/xd75/keymaps/daniel/rules.mk [hunk 12]
```diff
@@ -0,0 +1,3 @@
+ifndef QUANTUM_DIR
```
Comment (drashna) [inline keyboards/xd75/keymaps/daniel/rules.mk]: Could you remove this (ifndef/include/endif) block? 

It's not needed, and is a holdover from the old build system.

File: keyboards/xd75/keymaps/jarred/config.h [hunk 13]
```diff
@@ -0,0 +1,8 @@
+#ifndef CONFIG_USER_H
```
Comment (drashna) [inline keyboards/xd75/keymaps/jarred/config.h]: If you could replace the include guard (the ifndef, define, and endif at the end) with just `#pragma once`? 

It's simpler, and less prone to user error, so if you could make this switch?

File: keyboards/xd75/keymaps/jarred/config.h [hunk 14]
```diff
@@ -0,0 +1,8 @@
...
+#define CONFIG_USER_H
+
+#include "../../config.h"
```
Comment (drashna) [inline keyboards/xd75/keymaps/jarred/config.h]: This line is not actually needed, and should be removed. 

Specifically, the build script automatically includes all of the config.h files, so this is not necessary.

File: keyboards/xd75/keymaps/jarred/config.h [hunk 15]
```diff
@@ -0,0 +1,8 @@
...
+
+#define PREVENT_STUCK_MODIFIERS
```
Comment (drashna) [inline keyboards/xd75/keymaps/jarred/config.h]: This line is not actually needed, and should be removed.

Specifically, the build script automatically includes all of the config.h files, so this is not necessary.

File: users/jarred/config.h [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef USERSPACE_CONFIG_H
+#define USERSPACE_CONFIG_H
+
+#define PREVENT_STUCK_MODIFIERS
```
Comment (drashna) [inline users/jarred/config.h]: This line is not actually needed, and should be removed.

Specifically, the build script automatically includes all of the config.h files, so this is not necessary.

File: users/jarred/config.h [hunk 17]
```diff
@@ -0,0 +1,55 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef USERSPACE_CONFIG_H
```
Comment (drashna) [inline users/jarred/config.h]: If you could replace the include guard (the ifndef, define, and endif at the end) with just `#pragma once`? 

It's simpler, and less prone to user error, so if you could make this switch?

File: keyboards/xd75/keymaps/jarred/rules.mk [hunk 18]
```diff
@@ -0,0 +1,3 @@
+ifndef QUANTUM_DIR
```
Comment (drashna) [inline keyboards/xd75/keymaps/jarred/rules.mk]: Could you remove this (ifndef/include/endif) block? 

It's not needed, and is a holdover from the old build system.

File: keyboards/dz60/keymaps/jarred/config.h (hunk 1 above)
Comment (noroadsleft) [reply keyboards/dz60/keymaps/jarred/config.h]: @drashna Actually, I think this file can be deleted.

- Line 4 isn't needed; it's included automatically by the build script.
- Line 6 isn't needed. It's default behavior.

With your changes, that leaves only `#pragma once` in this file.

File: keyboards/satan/keymaps/jarred/config.h (hunk 5 above)
Comment (noroadsleft) [reply keyboards/satan/keymaps/jarred/config.h]: This file I believe can be deleted as well, for the same reasons as the `config.h` file for the DZ60 keymap.

File: keyboards/xd75/keymaps/daniel/config.h (hunk 9 above)
Comment (noroadsleft) [reply keyboards/xd75/keymaps/daniel/config.h]: This file I think can be deleted, as with the other keymap `config.h` files, for the same reasons as before.

File: keyboards/xd75/keymaps/daniel/rules.mk (hunk 12 above)
Comment (noroadsleft) [reply keyboards/xd75/keymaps/daniel/rules.mk]: This file should be deleted. The code block is unnecessary as @drashna said, and there's nothing else here.

File: keyboards/xd75/keymaps/jarred/config.h (hunk 13 above)
Comment (noroadsleft) [reply keyboards/xd75/keymaps/jarred/config.h]: @noroadsleft says "Delete this file, blah blah blah, I've heard it a hundred times already."

:grinning:

File: keyboards/xd75/keymaps/jarred/keymap.c [hunk 19]
```diff
@@ -0,0 +1,40 @@
+#include QMK_KEYBOARD_H
+
+#define _QW 0
+#define _LW 1
+#define _NV 2
+#define _NM 3
+
+#define _______ KC_TRNS
```
Comment (noroadsleft) [inline keyboards/xd75/keymaps/jarred/keymap.c]: Lines 8 and 9 can be safely deleted. QMK includes these definitions automatically with every keymap.

https://github.com/qmk/qmk_firmware/blob/4318797d198b58bb807b3e436c7d8924d8b4a6fe/quantum/quantum_keycodes.h#L29-L31

File: keyboards/xd75/keymaps/jarred/rules.mk (hunk 18 above)
Comment (noroadsleft) [reply keyboards/xd75/keymaps/jarred/rules.mk]: File can be deleted, nothing going on here.

**Comment by JarredSteenvoorden** [comment]
Wow there was alot of redundant code in these older keymaps, should be cleaned up now :)

File: users/jarred/jarred.h [hunk 20]
```diff
@@ -19,6 +19,10 @@
 
 #include "quantum.h"
 
+enum userspace_custom_keycodes {
+  VRSN = SAFE_RANGE               // Prints QMK Firmware and board info
+};
+
 // Use 7 wide characters for keymaps
```
Comment (noroadsleft) [inline users/jarred/jarred.h]: Missed this before.

Lines 26-28 shouldn't be needed.

But everything else to me looks good.

**Comment by drashna** [comment]
Thanks!

**Pull request merged**